core/client/fs/src/test/java/alluxio/client/file/cache/CacheManagerTest.java

```diff
@@ -28,6 +36,24 @@ public void factoryGet() throws Exception {
     assertEquals(manager, CacheManager.Factory.get(mConf));
   }
 
+  @Test
+  public void factoryGetConcurrently() throws Exception {
+    CacheManager.Factory.clear();
+    int cnt = 10;
+    List<Future<CacheManager>> cacheManagersList = new ArrayList<>(cnt);
+    ExecutorService executorService = Executors.newFixedThreadPool(cnt);
+    for (int i = 0; i < cnt; i++) {
+      cacheManagersList.add(executorService.submit(new CacheManagerFetch(mConf)));
```

Comment:
nit: you can replace this with lambda expression for readability, without introducing a new class `CacheManagerFetch`

```suggestion
      cacheManagersList.add(executorService.submit(() -> {
        try {
          return CacheManager.Factory.get(mConf);
        } catch (Exception e) {
          return null;
        }
      }));
```

Reply:
Done